content/en/blog/2022-11-16-templating-and-diffs-with-kustomize.md

```diff
@@ -0,0 +1,337 @@
+-rw-r--r--   1 user  wheel   558B Nov 16 16:28 configmap-discovery.yaml
+-rw-r--r--   1 user  wheel   1.5K Nov 16 16:28 deployment-entry.yaml
+-rw-r--r--   1 user  wheel   1.1K Nov 16 16:28 deployment-hat.yaml
+-rw-r--r--   1 user  wheel   1.1K Nov 16 16:28 deployment-left-arm.yaml
+-rw-r--r--   1 user  wheel   1.1K Nov 16 16:28 deployment-left-leg.yaml
+-rw-r--r--   1 user  wheel   1.1K Nov 16 16:28 deployment-right-arm.yaml
+-rw-r--r--   1 user  wheel   1.1K Nov 16 16:28 deployment-right-leg.yaml
+-rw-r--r--   1 user  wheel   474B Nov 16 16:28 kustomization.yaml
+-rw-r--r--   1 user  wheel   447B Nov 16 16:28 service-entry.yaml
+-rw-r--r--   1 user  wheel   438B Nov 16 16:28 service-hat.yaml
+-rw-r--r--   1 user  wheel   454B Nov 16 16:28 service-left-arm.yaml
+-rw-r--r--   1 user  wheel   453B Nov 16 16:28 service-left-leg.yaml
+-rw-r--r--   1 user  wheel   456B Nov 16 16:28 service-right-arm.yaml
+-rw-r--r--   1 user  wheel   456B Nov 16 16:28 service-right-leg.yaml
+-rw-r--r--   1 user  wheel   281B Nov 16 16:28 serviceaccount.yaml
+```
+
+As you can see, this is a simple Kustomize deployment, not using any bases or overlays. Let's start using Kluctl by
+doing a vanilla deployment first:
+
+```shell
+$ kluctl deploy
+✓ Loading kluctl project
+✓ Initializing k8s client
+✓ Rendering templates
+✓ Rendering Helm Charts
+✓ Building kustomize objects
+✓ Postprocessing objects
+✓ Getting namespaces
+✓ .: Applied 14 objects.
+✓ Running server-side apply for all objects
+
+New objects:
+  default/ConfigMap/podtato-head-service-discovery
+  default/Deployment/podtato-head-entry
+  default/Deployment/podtato-head-hat
+  default/Deployment/podtato-head-left-arm
+  default/Deployment/podtato-head-left-leg
+  default/Deployment/podtato-head-right-arm
+  default/Deployment/podtato-head-right-leg
+  default/Service/podtato-head-entry
+  default/Service/podtato-head-hat
+  default/Service/podtato-head-left-arm
+  default/Service/podtato-head-left-leg
+  default/Service/podtato-head-right-arm
+  default/Service/podtato-head-right-leg
+  default/ServiceAccount/podtato-head
+? The diff succeeded, do you want to proceed? (y/N) y
+✓ .: Applied 14 objects.
+✓ Running server-side apply for all objects
+
+New objects:
+  default/ConfigMap/podtato-head-service-discovery
+  default/Deployment/podtato-head-entry
+  default/Deployment/podtato-head-hat
+  default/Deployment/podtato-head-left-arm
+  default/Deployment/podtato-head-left-leg
+  default/Deployment/podtato-head-right-arm
+  default/Deployment/podtato-head-right-leg
+  default/Service/podtato-head-entry
+  default/Service/podtato-head-hat
+  default/Service/podtato-head-left-arm
+  default/Service/podtato-head-left-leg
+  default/Service/podtato-head-right-arm
+  default/Service/podtato-head-right-leg
+  default/ServiceAccount/podtato-head
+```
+
+It will first show you what it will do, then ask for confirmation and finally show what it did. Verify that it got
```

Comment:
```suggestion
`kluctl` shows you the diff it will apply, asks for confirmation, applies the changes and then shows you the applied result. Verify that it got
```